recipe/meta.yaml

```
@@ -1,67 +1,157 @@
+      commands:
+        - test ! -f ${PREFIX}/{{ lib_prefix }}libz.a            # [unix]
+        - test ! -f ${PREFIX}/{{ lib_prefix }}libz${SHLIB_EXT}  # [unix]
+        - test ! -f ${PREFIX}/include/zlib.h                    # [unix]
+        - if not exist %LIBRARY_BIN%\zlib.dll exit 1            # [win]
+        - if not exist %PREFIX%\zlib.dll exit 1                 # [win]
+
+  - name: lib{{ name }}-wapi
     build:
+      skip: true  # [not win]
+      run_exports:
+        - {{ pin_subpackage('lib' + name + '-wapi', max_pin='x') }}
+    requirements:
+      build:
+        - {{ stdlib('c') }}
         - {{ compiler('c') }}
-        - {{ compiler('cxx') }}  # [win32]
-        # Require `cmake-no-system` to break circular dependency;
-        # the `cmake` package on defaults requires `zstd`.
-        - cmake-no-system
-        - msinttypes  # [win and vc<14]
-        - make  # [not win]
-        - patch  # [not win]
-        - m2-patch  # [win]
-    host:
-        - ripgrep         # Should this be removed?
+      run_constrained:
+        - {{ name }} {{ version }}
+        - {{ name }}-wapi {{ version }}
+    files:
+      - {{ lib_prefix }}bin/zlibwapi.dll  # [win]
+    test:
+      commands:
+        - if not exist "%LIBRARY_BIN%\zlibwapi.dll" exit 1  # [win]
 
-test:
-    commands:
-       - test -f ${PREFIX}/include/zlib.h  # [unix]
-       - test -f ${PREFIX}/lib/libz.a      # [unix]
-       - test -f ${PREFIX}/lib/libz.so     # [linux]
-       - test -f ${PREFIX}/lib/libz.dylib  # [osx]
-       - if not exist %PREFIX%/Library/include/zlib.h exit 1         # [win]
-       - if not exist %PREFIX%/Library/lib/z.lib exit 1              # [win]
-       - if not exist %PREFIX%/Library/lib/zdll.lib exit 1           # [win]
-       - if not exist %PREFIX%/Library/lib/zlib.lib exit 1           # [win]
-       - if not exist %PREFIX%/Library/lib/zlibstatic.lib exit 1     # [win]
-       - if not exist %PREFIX%/Library/bin/zlib.dll exit 1           # [win]
+  - name: {{ name }}
+    build:
+      run_exports:
+        - {{ pin_subpackage('lib' + name, max_pin='x') }}
+    requirements:
+      build:
+        - {{ stdlib('c') }}
+        - {{ compiler('c') }}
+      host:
+        - {{ pin_subpackage('lib' + name, exact=True) }}
+      run:
+        - {{ pin_subpackage('lib' + name, exact=True) }}
+    files:
+      - {{ lib_prefix }}libz.so             # [linux]
+      - {{ lib_prefix }}libz.dylib          # [osx]
+      - include                             # [unix]
+      - {{ lib_prefix }}pkgconfig           # [unix]
+      - {{ lib_prefix }}libz.a              # [unix]
+      - {{ lib_prefix }}include             # [win]
+      - {{ lib_prefix }}share               # [win]
+      - {{ lib_prefix }}lib/zlib.lib        # [win]
+      - {{ lib_prefix }}lib/zdll.lib        # [win]
+      - {{ lib_prefix }}lib/z.lib           # [win]
+      - {{ lib_prefix }}lib/zlibstatic.lib  # [win]
+      - {{ lib_prefix }}lib/pkgconfig       # [win]
```

```suggestion
    files:
      - {{ lib_prefix }}lib/libz{{ SHLIB_EXT }}  # [unix]
      - {{ lib_prefix }}lib/libz.a               # [unix]
      - {{ lib_prefix }}lib/zlib.lib             # [win]
      - {{ lib_prefix }}lib/zdll.lib             # [win]
      - {{ lib_prefix }}lib/z.lib                # [win]
      - {{ lib_prefix }}lib/zlibstatic.lib       # [win]
      - {{ lib_prefix }}include
      - {{ lib_prefix }}lib/pkgconfig
      - {{ lib_prefix }}share
```
This would be one way to actually use those prefix variables you defined to some effect, assuming they were set to "Library" on Windows and nothing on Unix. If you leave these all listed in a per-platform way then the variables may as well not be replacing the text that was there previously since they all would evaluate the same.

For future reference, including top-level folders like `include` is dangerous if any dependencies are installed because it literally includes everything found in the host environment's include folder.

---

Made paths hardcoded